tests/test_definitions/test_expectations_cfe.py

```diff
@@ -178,100 +196,105 @@ def pytest_generate_tests(metafunc):
                         if not generate_test:
                             continue
 
-                        if "suppress_test_for" in test and (
-                            (
-                                "sqlalchemy" in test["suppress_test_for"]
-                                and validator_with_data
-                                and isinstance(
-                                    validator_with_data.execution_engine.active_batch_data,
-                                    SqlAlchemyBatchData,
-                                )
-                            )
-                            or (
-                                "sqlite" in test["suppress_test_for"]
-                                and sqliteDialect is not None
-                                and validator_with_data
-                                and isinstance(
-                                    validator_with_data.execution_engine.active_batch_data,
-                                    SqlAlchemyBatchData,
-                                )
-                                and isinstance(
-                                    validator_with_data.execution_engine.active_batch_data.sql_engine_dialect,
-                                    sqliteDialect,
-                                )
-                            )
-                            or (
-                                "postgresql" in test["suppress_test_for"]
-                                and postgresqlDialect is not None
-                                and validator_with_data
-                                and isinstance(
-                                    validator_with_data.execution_engine.active_batch_data,
-                                    SqlAlchemyBatchData,
+                        if "suppress_test_for" in test:
+                            if not isinstance(test["suppress_test_for"], list):
+                                raise ValueError(
+                                    "Invalid test specification. 'suppress_test_for' should be passed in as a list"
                                 )
```

Comment:
We weren't doing the same check for the `suppress_test_for` flag at all, so this was added